applications/GeoMechanicsApplication/custom_elements/geo_linear_truss_element.cpp

```diff
@@ -118,7 +118,7 @@ void GeoLinearTrussElement<TDim,TNumNodes>::
     }
 
     if (rVariable == FORCE) {
-        BoundedVector<double, TDim> truss_forces = ZeroVector(TDim);
+        BoundedVector<double, 3> truss_forces = ZeroVector(3);
```

Comment:
Are the results always 3 dimensional even in the 2D case ? As this is a templated class. If that is the case, is the same true in other locations where TDim is used.  Perhaps we can remove the TDim templating (and hardcode 3 in more places) of this and the geo_cable_element, if they have the same structure structures regardless of dimension size. Also do we plan to have more than 2 noded versions of the 2 elements ?  (I don't quite see why we would need it)

Reply:
Although, I guess length is an issue, but can't remember if the node points are still stored in 3D, if they are then that is equally not an issue

BTW this is just a discussion point and not 100% needed for the review

Reply:
truss_forces is projected on rOuput which is a vector of a array_1d with size 3. To prevent memory issues, TDim in this case had to be changed to 3